common.go

```diff
@@ -434,3 +441,16 @@ func GetLanguageByAlias(alias string) (lang string, ok bool) {
 
 	return
 }
+
+// GetLanguagesByGitattributes returns a length 1 slice with the language assigned in .gitattributes if the regular expresion
+// matchs with the filename. It is comply with the signature to be a Strategy type.
```

Comment:
Note that it returns an empty slice if the filename does not match any of the rules in .gitattributes.  Should this return nil instead?